gsc/_gsclib.scm

```diff
@@ -159,82 +200,83 @@
-              (if (##eq? ld-options-prelude (macro-absent-obj))
-                  '()
-                  (macro-force-vars (ld-options-prelude)
-                    ld-options-prelude)))
-             (ld-opts
-              (if (##eq? ld-options (macro-absent-obj))
-                  '()
-                  (macro-force-vars (ld-options)
-                    ld-options)))
-             (pkg-cfg
-              (if (##eq? pkg-config (macro-absent-obj))
-                  '()
-                  (macro-force-vars (pkg-config)
-                    pkg-config)))
-             (pkg-cfg-path
-              (if (##eq? pkg-config-path (macro-absent-obj))
-                  '()
-                  (macro-force-vars (pkg-config-path)
-                    pkg-config-path)))
-             (filename-or-source
-              (if (##eq? expression (macro-absent-obj))
-                  filename
-                  (if (##source? expression)
-                      expression
-                      (##make-source
-                       expression
-                       (##make-locat (##path->container filename) 0))))))
-        (cond ((##not (or (##null? opts)
-                          (##pair? opts)))
-               (error "list expected for options: parameter")) ;;;;;;;
-              ((##not (##string? out))
-               (error "string expected for output: parameter")) ;;;;;;;;;;
-              ((##not (or (##string? bas) (##eq? bas #f)))
-               (error "string or #f expected for base: parameter")) ;;;;;;;;;;
-              ((##not (##string-or-string-list? cc-opts))
-               (error "string or string list expected for cc-options: parameter")) ;;;;;;;;;;
-              ((##not (##string-or-string-list? ld-opts-prelude))
-               (error "string or string list expected for ld-options-prelude: parameter")) ;;;;;;;;;;
-              ((##not (##string-or-string-list? ld-opts))
-               (error "string or string list expected for ld-options: parameter")) ;;;;;;;;;;
-              ((##not (##string-or-string-list? pkg-cfg))
-               (error "string or string list expected for pkg-config: parameter")) ;;;;;;;;;;
-              ((##not (##string-or-string-list? pkg-cfg-path))
-               (error "string or string list expected for pkg-config-path: parameter")) ;;;;;;;;;;
-              (else
-               (##compile-file filename-or-source
-                               opts
-                               out
-                               bas
-                               cc-opts
-                               ld-opts-prelude
-                               ld-opts
-                               pkg-cfg
-                               pkg-cfg-path)))))))
+      (##with-resolved-input-file
+       filename
+       (if (##eq? options (macro-absent-obj))
+           '()
+           (macro-force-vars (options)
+             options))
+       (lambda (filename target-dir opts)
+         (let* ((out
```

Comment:
The changes here are mostly indentation to account for wrapping everything in a lambda.